### PR TITLE
Feature/login button

### PR DESCRIPTION
**Pull request by hborrelli1**
## Type of change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Styling- no new features
- [ ] Refactoring- no new features

## Detailed Description
- Change password input validation
- Removed default state for userInfo outlined in App and moved the functionality to the userInfo reducer.
- Add nav welcome message if logged in.
- Fix routing for login button depending on logged in status.
- Add error message if email or password are invalid. 
- Styled login component.

## Why is this change required? What problem does it solve?
- cleans up form validation for standard email structure. Once the user logs in or out they are redirected to the homepage.

## Files modified:
- src/actions/index.js
- src/components/App/App.js
- src/components/App/App.scss
- src/components/Login/Login.js
- src/components/Login/Login.scss
- src/components/MovieContainer/MovieContainer.js
- src/components/Nav/Nav.js
- src/components/Nav/Nav.scss
- src/reducers/userInfo.js 
- src/sass/_mixins.scss
- src/sass/_variables.scss
- src/sass/application.scss

### CODE REVIEW

File: src/components/Login/Login.js [hunk 1]
```diff
@@ -60,7 +60,12 @@ class Login extends Component {
 
   validateForm = () => {
     const { email, password } = this.state;
-    const validEmail = email.includes('@') && email.includes('.io')
+    const validEmailRegex = RegExp(
+      // eslint-disable-next-line
+      /^(([^<>()\[\]\.,;:\s@\']+(\.[^<>()\[\]\.,;:\s@\']+)*)|(\'.+\'))@(([^<>()[\]\.,;:\s@\']+\.)+[^<>()[\]\.,;:\s@\']{2,})$/i
```
Comment (megan-venetianer) [inline src/components/Login/Login.js:65]: Would love to get an explanation on this!

File: src/components/Login/Login.js [hunk 2]
```diff
@@ -34,10 +34,14 @@ class Login extends Component {
             })
         } else {
           return response.json()
-            .then(data => this.setState({ error:data.error }))
+            .then(data => {
+              this.setState({ error:data.error })
+              console.log(data.error);
```
Comment (CLPena) [inline src/components/Login/Login.js:39]: This is interesting. Is there a purpose to console logging the error here?

File: src/components/Login/Login.scss [hunk 3]
```diff
@@ -0,0 +1,48 @@
+.login-container {
```
Comment (CLPena) [inline src/components/Login/Login.scss:1]: This all looks great. We just want to make sure to alphabetize properties before submitting our project (I have an atom package that can do it automatically if you want me to just check that at the end)

**Comment by CLPena** [comment]
Looks like you got a lot done! Thanks for working so hard on this. I left a few comments, but nothing urgent.

File: src/components/Login/Login.js [hunk 4]
```diff
@@ -70,30 +79,30 @@ class Login extends Component {
...
       <form>
         <h1>Login</h1>
         <span className="error">{this.state.error}</span>
         <label htmlFor="email">Email</label>
         <input
           type="text"
           name="email"
-          placeholder="enter email"
+          placeholder="Enter email"
           onChange={this.handleUpdate}
```
Comment (TimNguyen21) [inline src/components/Login/Login.js:91]: These name changes may break the Login.test, but it's ok. we can fix it in the next PR.

**Comment by TimNguyen21** [comment]
The PR looks good where it adds additional login functionality to the app.

**Pull request merged**